pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go

```diff
@@ -144,26 +145,32 @@ func (cra *cpuResourceAdvisor) GetHeadroom() (resource.Quantity, error) {
 		return *resource.NewQuantity(int64(cra.metaServer.NumCPUs-reservePoolSize), resource.DecimalSI), nil
 	}
 
-	hasShareRegion := false
+	shareRegionRequirement := 0
+
 	totalHeadroom := resource.NewQuantity(0, resource.DecimalSI)
 	for _, r := range cra.regionMap {
+		if r.Type() == types.QoSRegionTypeShare {
```

Comment:
I have had an in-depth discussion with @sun-yuliang  on this topic. If we want to decouple the calculation headroom of each share region, we need to set a specific total value for each region. However, how to divide the total value is a difficult problem. It may be possible to solve it through multiple iterations, but it is not a problem that we need to solve in the short term. That's why we wrote this rather ugly but logically correct code